### PR TITLE
Return slice into buffer when parsing byte vector

### PR DESCRIPTION
**Pull request by snosenzo**
- return slice into buffer when parsing byte vector
- addresses: https://github.com/jkuszmaul/flatbuffers_reflection/issues/22

### CODE REVIEW

**Comment by jkuszmaul** [comment]
Huh, my CI doesn't work. That's not your problem; I'll worry about it later.

If it's not too hard to add a test to the reflection.test.ts for this, I'd appreciate it (otherwise I'll add one later). Probably means copying over a .fbs with a byte vector field and generated .bfbs and _generated.ts and checking that.

File: src/reflection.ts [hunk 1]
```diff
@@ -428,21 +425,32 @@ export class Parser {
     if (fieldType.baseType() !== reflection.BaseType.Vector) {
       throw new Error("Field " + fieldName + " is not an vector.");
     }
-    if (!isScalar(fieldType.element())) {
+    const elementType = fieldType.element();
+    if (!isScalar(elementType)) {
       throw new Error("Field " + fieldName + " is not an vector of scalars.");
     }
+    const isByteVector =
+      elementType === reflection.BaseType.UByte || elementType === reflection.BaseType.Byte;
```
Comment (jkuszmaul) [inline src/reflection.ts:433]: I'm not too familiar with type/javascript conventions around this, but is there a reason not to use Int8Array for the signed Byte?

Comment (snosenzo) [reply src/reflection.ts:433]: yeah, was trying to cover your comment (vector of bytes), but no real reason other than that. I can scope it to just use uint8array for ubyte.

**Comment by jkuszmaul** [comment]
Thank you! I'll get this published at some point shortly (hopefully just later today).

**Pull request merged**